command/sidechain/enlisting/enlist_validator.go

```diff
@@ -0,0 +1,133 @@
+package enlisting
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/command"
+	"github.com/0xPolygon/polygon-edge/command/helper"
+	"github.com/0xPolygon/polygon-edge/command/polybftsecrets"
+	sidechainHelper "github.com/0xPolygon/polygon-edge/command/sidechain"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/contracts"
+	"github.com/0xPolygon/polygon-edge/txrelayer"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/spf13/cobra"
+	"github.com/umbracle/ethgo"
+)
+
+var (
+	stakeManager      = contracts.ValidatorSetContract
+	whitelistFn       = contractsapi.ChildValidatorSet.Abi.Methods["addToWhitelist"]
+	whitelistEventABI = contractsapi.ChildValidatorSet.Abi.Events["AddedToWhitelist"]
+)
+
+var params enlistParams
+
+func GetCommand() *cobra.Command {
+	registerCmd := &cobra.Command{
+		Use:     "enlist-validator",
+		Short:   "enlist a new validator",
+		PreRunE: runPreRun,
+		RunE:    runCommand,
+	}
+
+	setFlags(registerCmd)
+
+	return registerCmd
+}
+
+func setFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVar(
+		&params.accountDir,
+		polybftsecrets.DataPathFlag,
+		"",
+		polybftsecrets.DataPathFlagDesc,
+	)
+
+	cmd.Flags().StringVar(
+		&params.configPath,
+		polybftsecrets.ConfigFlag,
+		"",
+		polybftsecrets.ConfigFlagDesc,
+	)
+
+	cmd.Flags().StringVar(
+		&params.newValidatorAddress,
+		newValidatorAddressFlag,
+		"",
+		"account address of a possible validator",
+	)
+
+	cmd.MarkFlagsMutuallyExclusive(polybftsecrets.DataPathFlag, polybftsecrets.ConfigFlag)
+	helper.RegisterJSONRPCFlag(cmd)
+}
+
+func runPreRun(cmd *cobra.Command, _ []string) error {
+	params.jsonRPC = helper.GetJSONRPCAddress(cmd)
+
+	return params.validateFlags()
+}
+
+func runCommand(cmd *cobra.Command, _ []string) error {
+	outputter := command.InitializeOutputter(cmd)
+	defer outputter.WriteOutput()
+
+	ownerAccount, err := sidechainHelper.GetAccount(params.accountDir, params.configPath)
+	if err != nil {
+		return fmt.Errorf("enlist validator failed: %w", err)
+	}
+
+	inputAddress := types.StringToAddress(params.newValidatorAddress)
+
+	txRelayer, err := txrelayer.NewTxRelayer(txrelayer.WithIPAddress(params.jsonRPC),
+		txrelayer.WithReceiptTimeout(150*time.Millisecond))
+	if err != nil {
+		return fmt.Errorf("enlist validator failed: %w", err)
+	}
+
+	encoded, err := whitelistFn.Encode([]interface{}{
```

Comment:
You can include this function in `contractsapi` to be generated as a Go struct, and use it here. Then you can remove that `whitelistFn` variable.

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1258/commits/323027a56e7b07a3eb9c9c25816e8226bbcfee5a